crates/db_schema/src/impls/actor_language.rs

```diff
@@ -0,0 +1,437 @@
+        community_language
+          .filter(language_id.eq(for_language_id))
+          .filter(community_id.eq(for_community_id)),
+      ))
+      .get_result(conn)?;
+
+      if is_allowed {
+        Ok(())
+      } else {
+        Err(LemmyError::from_message("language_not_allowed"))
+      }
+    } else {
+      Ok(())
+    }
+  }
+
+  /// When site languages are updated, delete all languages of local communities which are not
+  /// also part of site languages. This is because post/comment language is only checked against
+  /// community language, and it shouldnt be possible to post content in languages which are not
+  /// allowed by local site.
+  fn limit_languages(conn: &mut PgConnection) -> Result<(), Error> {
+    use crate::schema::{
+      community::dsl as c,
+      community_language::dsl as cl,
+      site_language::dsl as sl,
+    };
+    let community_languages: Vec<LanguageId> = cl::community_language
+      .left_outer_join(sl::site_language.on(cl::language_id.eq(sl::language_id)))
+      .inner_join(c::community)
+      .filter(c::local)
+      .filter(sl::language_id.is_null())
+      .select(cl::language_id)
+      .get_results(conn)?;
+
+    for c in community_languages {
+      delete(cl::community_language.filter(cl::language_id.eq(c))).execute(conn)?;
+    }
+    Ok(())
+  }
+
+  pub fn read(
+    conn: &mut PgConnection,
+    for_community_id: CommunityId,
+  ) -> Result<Vec<LanguageId>, Error> {
+    use crate::schema::community_language::dsl::*;
+    community_language
+      .filter(community_id.eq(for_community_id))
+      .select(language_id)
+      .get_results(conn)
+  }
+
+  pub fn update(
+    conn: &mut PgConnection,
+    mut language_ids: Vec<LanguageId>,
+    for_community_id: CommunityId,
+  ) -> Result<(), Error> {
+    conn.build_transaction().read_write().run(|conn| {
+      use crate::schema::community_language::dsl::*;
+      // Clear the current languages
+      delete(community_language.filter(community_id.eq(for_community_id))).execute(conn)?;
+
+      if language_ids.is_empty() {
+        language_ids = SiteLanguage::read_local(conn)?;
+      }
+      for l in language_ids {
+        let form = CommunityLanguageForm {
+          community_id: for_community_id,
+          language_id: l,
+        };
+        insert_into(community_language)
+          .values(form)
+          .get_result::<Self>(conn)?;
+      }
+      Ok(())
+    })
+  }
+}
+
+pub fn default_post_language(
+  conn: &mut PgConnection,
+  community_id: CommunityId,
+  local_user_id: LocalUserId,
+) -> Result<Option<LanguageId>, Error> {
+  use crate::schema::{community_language::dsl as cl, local_user_language::dsl as ul};
+  let intersection = ul::local_user_language
+    .inner_join(cl::community_language.on(ul::language_id.eq(cl::language_id)))
+    .filter(ul::local_user_id.eq(local_user_id))
+    .filter(cl::community_id.eq(community_id))
+    .select(cl::language_id)
+    .get_results::<LanguageId>(conn)?;
```

Comment:
Perfect. I think I need some way to get this info to the front end tho.

Reply:
Added this to GetCommunityView.